ui/src/components/connections/ConnectionDeleteModal.tsx

```diff
@@ -0,0 +1,78 @@
+import { Checkbox, Divider, Modal, Space, Typography, message } from "antd";
+import { CheckboxChangeEvent } from "antd/es/checkbox";
+import { useState } from "react";
+import { deleteConnection } from "src/adapters/api/connections";
+
+import { ReactComponent as IconClose } from "src/assets/icons/x.svg";
+import { useEnvContext } from "src/contexts/env";
+
+export function ConnectionDeleteModal({
+  id,
+  onClose,
+  onDelete,
+  open = true,
```

Comment:
I don't think we need this prop. We can pass always true and render the component conditionally.

Reply:
I thought about it in the beginning but with this approach, we are not mounting and unmounting the component in `ConnectionDetails`